hack/lib/golang.sh

```diff
@@ -569,14 +569,25 @@ kube::golang::setup_env() {
   export GO15VENDOREXPERIMENT=1
 }
 
+# kube::golang::hack_tools_gotoolchain outputs the value to use for $GOTOOLCHAIN,
```

Comment:
This tells us what it does, but not when/why to use it?
We should probably mention the intent to be used for compiling development tools versus the project sources.

Reply:
done